doc/source/conf.py

```diff
@@ -226,20 +227,69 @@
+
 moved_api_pages = [
-    'pandas.core.common.isnull', 'pandas.core.common.notnull', 'pandas.core.reshape.get_dummies',
-    'pandas.tools.merge.concat', 'pandas.tools.merge.merge', 'pandas.tools.pivot.pivot_table',
-    'pandas.tseries.tools.to_datetime', 'pandas.io.clipboard.read_clipboard', 'pandas.io.excel.ExcelFile.parse',
-    'pandas.io.excel.read_excel', 'pandas.io.html.read_html', 'pandas.io.json.read_json',
-    'pandas.io.parsers.read_csv', 'pandas.io.parsers.read_fwf', 'pandas.io.parsers.read_table',
-    'pandas.io.pickle.read_pickle', 'pandas.io.pytables.HDFStore.append', 'pandas.io.pytables.HDFStore.get',
-    'pandas.io.pytables.HDFStore.put', 'pandas.io.pytables.HDFStore.select', 'pandas.io.pytables.read_hdf',
-    'pandas.io.sql.read_sql', 'pandas.io.sql.read_frame', 'pandas.io.sql.write_frame',
-    'pandas.io.stata.read_stata']
-
-html_additional_pages = {'generated/' + page: 'api_redirect.html' for page in moved_api_pages}
+    ('pandas.core.common.isnull', 'pandas.isnull'),
+    ('pandas.core.common.notnull', 'pandas.notnull'),
+    ('pandas.core.reshape.get_dummies', 'pandas.get_dummies'),
+    ('pandas.tools.merge.concat', 'pandas.concat'),
+    ('pandas.tools.merge.merge', 'pandas.merge'),
+    ('pandas.tools.pivot.pivot_table', 'pandas.pivot_table'),
+    ('pandas.tseries.tools.to_datetime', 'pandas.to_datetime'),
+    ('pandas.io.clipboard.read_clipboard', 'pandas.read_clipboard'),
+    ('pandas.io.excel.ExcelFile.parse', 'pandas.ExcelFile.parse'),
+    ('pandas.io.excel.read_excel', 'pandas.read_excel'),
+    ('pandas.io.html.read_html', 'pandas.read_html'),
+    ('pandas.io.json.read_json', 'pandas.read_json'),
+    ('pandas.io.parsers.read_csv', 'pandas.read_csv'),
+    ('pandas.io.parsers.read_fwf', 'pandas.read_fwf'),
+    ('pandas.io.parsers.read_table', 'pandas.read_table'),
+    ('pandas.io.pickle.read_pickle', 'pandas.read_pickle'),
+    ('pandas.io.pytables.HDFStore.append', 'pandas.HDFStore.append'),
+    ('pandas.io.pytables.HDFStore.get', 'pandas.HDFStore.get'),
+    ('pandas.io.pytables.HDFStore.put', 'pandas.HDFStore.put'),
+    ('pandas.io.pytables.HDFStore.select', 'pandas.HDFStore.select'),
+    ('pandas.io.pytables.read_hdf', 'pandas.read_hdf'),
+    ('pandas.io.sql.read_sql', 'pandas.read_sql'),
+    ('pandas.io.sql.read_frame', 'pandas.read_frame'),
+    ('pandas.io.sql.write_frame', 'pandas.write_frame'),
+    ('pandas.io.stata.read_stata', 'pandas.read_stata'),
+]
+
+# Again, tuples of (from_old, to_new)
+moved_classes = [
+    ('pandas.tseries.resample.Resampler', 'pandas.core.resample.Resampler'),
+    ('pandas.formats.style.Styler', 'pandas.io.formats.style.Styler'),
+]
+
+for old, new in moved_classes:
+    # the class itself...
+    moved_api_pages.append((old, new))
+
+    mod, classname = new.rsplit('.', 1)
+    klass = getattr(importlib.import_module(mod), classname)
+    methods = [x for x in dir(klass)
+               if not x.startswith('_') or x.startswith('__')]
```

Comment:
the second is to *include* the `__` methods. We have http://pandas.pydata.org/pandas-docs/stable/generated/pandas.tseries.resample.Resampler.__iter__.html

I could just whitelist `__iter__`... We document that and `__array__` on Categorical.